### DISCUSSION TITLE
Are resources only configurable at the Component level?

### DISCUSSION THREAD

**Discussion by dtanner** [redpanda-data/connect#1514] [Q&A] [answered]
Hi - New user question...it appears I can configure Resources at the component level, and not any other granularity. Is this correct?
e.g. I could create a resource for a `kafka` input, but I can't create a resource for the `tls` section within kafka. 
(That's my use case; there's a lot configuration that's the same for the input and I just need to toggle TLS settings between local and deployed environments.)

Thanks!

**Comment by mihaitodor** [comment, marked as answer]
Hey, you can use yaml anchors to DRY those up: https://ktomk.github.io/writing/yaml-anchor-alias-and-merge-key.html However, if it becomes too unwieldy, you might want to try switching to CUE: https://www.benthos.dev/docs/configuration/using_cue

**Reply by dtanner** [reply]
Perfect - thanks @mihaitodor !
